### PR TITLE
Refactor listunspent

### PR DESCRIPTION
**Pull request by drgora**
This PR refactors the `listunspent` function, and its callees, to improve its performance in terms of execution time.

### CODE REVIEW

File: qa/pull-tester/rpc-tests.sh [hunk 1]
```diff
@@ -169,6 +169,7 @@ testScripts=(
   'mempool_hard_fork_cleaning.py',156,344
   'shieldedpoolremoval.py',377,855
   'sc_mempool_conflict.py',32,45
+  'listunspent.py',26,41
```
Comment (drgora) [inline qa/pull-tester/rpc-tests.sh:172]: Leaving a reminder for myself: update the release notes!

**Review by ptagl — APPROVED** [review]
It looks very good to me, thanks a lot!

File: src/univalue/include/univalue.h [hunk 2]
```diff
@@ -99,6 +99,7 @@ class UniValue {
 
     bool push_back(const UniValue& val);
     bool push_backV(const std::vector<UniValue>& vec);
+    UniValue& emplace_back(VType type) { values.emplace_back(type); return values.back(); };
```
Comment (JackPiri) [inline src/univalue/include/univalue.h:103]: We need to update the header for this file.

**Review by JackPiri — APPROVED** [review]
The PR is fine to me.
Thanks

File: src/univalue/include/univalue.h (hunk 2 above)
Comment (drgora) [reply src/univalue/include/univalue.h:103]: Good point, done, thanks!

**Review by 95DDB — APPROVED** [review]

**Pull request merged**